src/Symfony/Component/Asset/Context/ContextFactory.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Asset\Context;
+
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Routing\RequestContext;
+
+/**
+ * @author Roland Franssen <franssen.roland@gmail.com>
+ */
+class ContextFactory
+{
+    private $requestStack;
+    private $requestContext;
+
+    public function __construct(RequestStack $requestStack = null, RequestContext $requestContext = null)
+    {
+        $this->requestStack = $requestStack;
+        $this->requestContext = $requestContext;
+    }
+
+    /**
+     * @return ContextInterface
+     */
+    public function createContext()
+    {
+        if (null !== $this->requestStack && $this->requestStack->getMasterRequest()) {
```

Comment:
this is wrong, as the request may be injected in the request stack later than that. You cannot instantiate the service based on the availability of the Request, as it would change the behavior depending on when the context is instantiated (which depends on what needs it in the container and so is unknown to you)